src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java

```diff
@@ -108,6 +116,61 @@ public void onFailure()
       }
    }
    
+   public void interruptRNoDebug(final InterruptHandler handler) {
+      final int originalDebugType = errorManager_.getErrorHandlerType();
+      
+      if (originalDebugType != ErrorHandlerType.ERRORS_BREAK) {
```

Comment:
This catches the most common case but we might want to reset user-supplied handlers as well (e.g. `options(error = browser)` would cause trouble here)